src/transaction/log_manager.c

```diff
@@ -575,6 +583,37 @@ log_get_eof_lsa (void)
   return (&log_Gl.hdr.eof_lsa);
 }
 
+/*
+ * log_get_num_free_block -
+ *
+ * return:
+ *
+ * NOTE:
+ */
+void
+log_get_num_free_block (INT64 * num_free_block)
+{
+#if defined(SOLARIS)
+  struct statvfs buf;
+#else /* SOLARIS */
+  struct statfs buf;
+#endif /* SOLARIS */
+
+  memset (&buf, 0, sizeof (buf));
+  *num_free_block = 0;
+
+#if defined(SOLARIS)
+  if (statvfs (log_Path, &buf) == 0)
+#elif defined(AIX)
+  if (statfs ((char *) log_Path, &buf) == 0)
+#else /* AIX */
+  if (statfs (log_Path, &buf) == 0)
```

Comment:
Isn't it more accurate to judge by narrowing the error code to ENOMEM when performing statfs and failing?

Reply:
How about leaving syslog in this function if errno is ENOMEM or EIO?

Reply:
It's not bad. but the first idea was to let the cub_master know the information and cub_master leave the info to the log so finally make the 'admin' know the info from only logs by cub_master, isn't it?

Reply:
The cub_master may not be able to write an error log due to same problem of a file system or disk of a cub _server if errno is ENOMEM or EIO.